### PR TITLE
Add install script and readme snippet

### PR DESCRIPTION
**Pull request by cmoog**
Inspired by the simplicity of https://raw.githubusercontent.com/denoland/deno_install/master/install.sh

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# TODO(everyone): Keep this script simple and easily auditable.
+
+set -e
+
+if [ "$(uname -m)" != "x86_64" ]; then
+  echo "Error: Unsupported architecture $(uname -m). Only x64 binaries are available." 1>&2
+  exit 1
+fi
+
+if [ "$OS" = "Windows_NT" ]; then
```
Comment (nhooyr) [inline install.sh:11]: Does windows define `OS`?

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  target="windows-386"
+  ext=".zip"
+  if ! command -v unzip >/dev/null; then
+    echo "Error: unzip is required to install coder-cli" 1>&2
+    exit 1
+  fi
+else
+  if ! command -v tar >/dev/null; then
```
Comment (nhooyr) [inline install.sh:19]: Why is this required? Won't it just exit out with command not found?

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    exit 1
+  fi
+  ext=".tar.gz"
+  case $(uname -s) in
+  Darwin) target="darwin-amd64" ;;
+  *) target="linux-amd64" ;;
+  esac
+fi
+
+if [ $# -eq 0 ]; then
```
Comment (nhooyr) [inline install.sh:30]: Let's use a `$version` var for this to make it clear.

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  if [ ! "$coder_asset_path" ]; then
+    echo "Error: Unable to find latest coder-cli release on GitHub." 1>&2
+    exit 1
+  fi
+  cdr_uri="https://github.com${coder_asset_path}"
+else
+  cdr_uri="https://github.com/cdr/coder-cli/releases/download/${1}/coder-cli-${target}${ext}"
+fi
+
+coder_install="${CODER_INSTALL:-$HOME/.coder}"
```
Comment (nhooyr) [inline install.sh:45]: This should be `~/.local/lib/coder`.

See https://github.com/cdr/code-server/blob/11b51058355caa60d21f59b09996d127776c8c84/install.sh#L42-L46

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+bin_dir="$coder_install/bin"
+exe="$bin_dir/coder"
+
+if [ ! -d "$bin_dir" ]; then
+  mkdir -p "$bin_dir"
+fi
+
+curl --fail --location --progress-bar --output "$exe$ext" "$cdr_uri"
```
Comment (nhooyr) [inline install.sh:53]: Can we make the flag short here too?

Comment (nhooyr) [inline install.sh:53]: `exe` and `ext` are very confusing side by side

**Review by nhooyr — CHANGES_REQUESTED** [review]

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
+# TODO(everyone): Keep this script simple and easily auditable.
```
Comment (nhooyr) [inline install.sh:2]: I like this!

File: install.sh (hunk 2 above)
Comment (cmoog) [reply install.sh:19]: Much cleaner to handle gracefully with a specific error message.

File: install.sh (hunk 5 above)
Comment (cmoog) [reply install.sh]: Why though? Not sure why we'd intentionally make it harder to read 😆

File: install.sh (hunk 1 above)
Comment (cmoog) [reply install.sh:11]: Not quite sure, although I suspect the Deno authors have tested this properly. Either way, I'll put together a Github actions test for this that runs on windows.

File: install.sh (hunk 4 above)
Comment (cmoog) [reply install.sh:46]: Is there a specific standard for this that I'm unaware of? I find having these kinds of hidden directories in my home folder convenient. Especially if we add configuration files to this directory down the line.

File: install.sh (hunk 2 above)
Comment (nhooyr) [reply install.sh:19]: How is it cleaner? It bloats the install script when the shell itself already handles it thanks to `set -e`.

File: install.sh (hunk 4 above)
Comment (nhooyr) [reply install.sh:46]: Yes, the XDG standard. It's `~/.local`. And then `~/.config` for the configuration files.

File: install.sh (hunk 5 above)
Comment (nhooyr) [reply install.sh]: No one uses the long flags for `curl`, everyone knows the short flags so they're easier to read.

Comment (wbobeirne) [reply install.sh]: I do not know the long flags and I appreciate that moogster went long.

Comment (vapurrmaid) [reply install.sh]: Same here, I pretty much always use long flags wherever possible

Comment (nhooyr) [reply install.sh]: Then let's be consistent with the above invocation of curl.

File: install.sh (hunk 1 above)
Comment (cmoog) [reply install.sh:13]: Just verified this.

File: install.sh (hunk 4 above)
Comment (cmoog) [reply install.sh:48]: So there are zero programs on my machine in `~/.local/lib`. Inclined to ignore the standard.

**Comment by cmoog** [comment]
Opting for package managers instead.
